sdk/src/types/block/output/feature/metadata.rs

```diff
@@ -74,6 +74,333 @@ impl core::fmt::Debug for MetadataFeature {
+        /// Royalty payment addresses mapped to the payout percentage.
+        #[serde(default, skip_serializing_if = "BTreeMap::is_empty")]
+        royalties: BTreeMap<Bech32Address, f64>,
+        /// The human-readable name of the native token creator.
+        #[serde(default, skip_serializing_if = "Option::is_none")]
+        issuer_name: Option<String>,
+        /// The human-readable description of the token.
+        #[serde(default, skip_serializing_if = "Option::is_none")]
+        description: Option<String>,
+        /// Additional attributes which follow [OpenSea Metadata standards](https://docs.opensea.io/docs/metadata-standards).
+        #[serde(default, skip_serializing_if = "BTreeSet::is_empty")]
+        attributes: BTreeSet<Attribute>,
+    }
+
+    impl Irc27Metadata {
+        pub fn new(media_type: impl Into<String>, uri: Url, name: impl Into<String>) -> Self {
+            Self {
+                version: "v1.0".to_owned(),
+                media_type: media_type.into(),
+                uri,
+                name: name.into(),
+                collection_name: Default::default(),
+                royalties: Default::default(),
+                issuer_name: Default::default(),
+                description: Default::default(),
+                attributes: Default::default(),
+            }
+        }
+
+        pub fn with_collection_name(mut self, collection_name: impl Into<String>) -> Self {
+            self.collection_name.replace(collection_name.into());
+            self
+        }
+
+        pub fn add_royalty(mut self, address: Bech32Address, percentage: f64) -> Self {
+            self.royalties.insert(address, percentage);
+            self
+        }
+
+        pub fn with_royalties(mut self, royalties: BTreeMap<Bech32Address, f64>) -> Self {
+            self.royalties = royalties;
+            self
+        }
+
+        pub fn with_issuer_name(mut self, issuer_name: impl Into<String>) -> Self {
+            self.issuer_name.replace(issuer_name.into());
+            self
+        }
+
+        pub fn with_description(mut self, description: impl Into<String>) -> Self {
+            self.description.replace(description.into());
+            self
+        }
+
+        pub fn add_attribute(mut self, attribute: Attribute) -> Self {
+            self.attributes.insert(attribute);
+            self
+        }
+
+        pub fn with_attributes(mut self, attributes: BTreeSet<Attribute>) -> Self {
+            self.attributes = attributes;
+            self
+        }
+
+        pub fn to_bytes(&self) -> Vec<u8> {
+            serde_json::to_string(self).unwrap().into_bytes()
+        }
+    }
+
+    impl TryFrom<Irc27Metadata> for MetadataFeature {
+        type Error = Error;
+        fn try_from(value: Irc27Metadata) -> Result<Self, Error> {
+            Self::new(value.to_bytes())
+        }
+    }
+
+    #[derive(Clone, Debug, Serialize, Deserialize, Getters, PartialEq, Eq)]
+    #[getset(get = "pub")]
+
```

Comment:
sad format noises.
also, missing rename camelCase (not that its likely used outside of Irc27Metadata but still)

Reply:
Not camelCase here https://docs.opensea.io/docs/metadata-standards#attributes